### PR TITLE
update tidb roadmap

### PR DESCRIPTION
**Pull request by samhld**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [ ] master (the latest development version)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-roadmap.md
Comment (Oreoxmt) [inline tidb-roadmap.md:1]: Preview: https://pingcap-docs-website-scaffold-git-preview-pingca-285466-oreoxmt.vercel.app/tidb/dev/tidb-roadmap

**Comment by lilin90** [comment]
> [CLA assistant check](https://cla-assistant.io/pingcap/docs?pullRequest=15194) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/pingcap/docs?pullRequest=15194) before we can accept your contribution.**1** out of **2** committers have signed the CLA.✅ Oreoxmt❌ Sam Dillard
> 
> **Sam Dillard** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).
> You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/pingcap/docs?pullRequest=15194) it.

It seems that @samhld still need to sign the CLA.

**Comment by easonn7** [comment]
LGTM

File: tidb-roadmap.md [hunk 1]
```diff
@@ -62,6 +55,10 @@ In the course of development, this roadmap is subject to change based on user ne
             <b>Unlimited transaction size</b>
           </li>
           <br />
+          <li>
+            <b>Federated query</b>
+            <i>TiDB query planner support for multiple storage engines in HTAP use cases.</i>
```
Comment (Oreoxmt) [inline tidb-roadmap.md:60]:
```suggestion
            <br /><i>TiDB query planner support for multiple storage engines in HTAP use cases.</i>
```

**Review by Oreoxmt — APPROVED** [review]

File: tidb-roadmap.md [hunk 2]
```diff
@@ -28,29 +28,22 @@ In the course of development, this roadmap is subject to change based on user ne
       <td>
         <ul>
           <li>
-            <b>GA of Partitioned Raft KV storage engine</b><br /><i
-              >PB-scale clusters, increased write velocity, faster scaling operations, and improved compaction stability</i
-            >
+            <b>Distributed execution framework</b><br /><i>
+            In v7.2.0, TiDB introduced the distributed execution framework for background tasks (such as DDL and analyze). This is the foundation for parallelizing these tasks across compute nodes. v7.4.0 introduces global sorting in distributed re-organization tasks (such as DDL and import), which greatly mitigates extra resource consumption in storage. Optionally, external shared storage can be leveraged for simplicity and cost savings</i>
```
Comment (ran-huang) [inline tidb-roadmap.md:32]:
```suggestion
            In v7.2.0, TiDB introduced the distributed execution framework for background tasks (such as DDL and analyze). This is the foundation for parallelizing these tasks across compute nodes. v7.4.0 introduces global sorting in distributed re-organization tasks (such as DDL and import), which greatly mitigates extra resource consumption in storage. Optionally, external shared storage can be leveraged for simplicity and cost savings.</i>
```

File: tidb-roadmap.md [hunk 3]
```diff
@@ -79,63 +76,32 @@ In the course of development, this roadmap is subject to change based on user ne
               Control over how background tasks, such as imports, DDL, TTL, auto-analyze, and compactions, can affect foreground traffic
             </i>
           </li>
-        </ul>
-      </td>
-      <td>
-        <ul>
+          <br>
+          </br>
```
Comment (ran-huang) [inline tidb-roadmap.md:80]: We should use a consistent approach when writing self-closing tags.
```suggestion
          <br />
```

File: tidb-roadmap.md [hunk 4]
```diff
@@ -79,63 +76,32 @@ In the course of development, this roadmap is subject to change based on user ne
...
-          </li>
-          <br />
-          <li>
-            <b>User-defined functions</b>
+            <b>Multi-tenancy</b>
+            <br /><i>Resource isolation on top of resource control</i>
           </li>
-          <br />
+          <br>
+          </br>
```
Comment (ran-huang) [inline tidb-roadmap.md:104]:
```suggestion
          <br />
```

**Review by ran-huang — APPROVED** [review]

**Comment by Oreoxmt** [comment]
@samhld The [CLA check fails](https://github.com/pingcap/docs/pull/15194#issuecomment-1781448548) because the commit author's email (`Author: Sam Dillard <samdillard@Sams-MacBook-Pro.local>`) is invalid. Please update your local git configuration and correct the author information on the existing commits.

1. Update your local git configuration:

```bash
cd docs
git config user.email sam@pingcap.com
```

2. Correct the author information on the existing commits:

	```bash
	git rebase -i 6f9ef141791c8b6becf4ba32d1ffcfd428a744f6
	```

	In the text editor that opens, replace `pick` with `edit` next to commit `3a20a0653` and `2f2396499`:
	
	```
	edit 3a20a0653 update tidb-roadmap.md -- first draft
	pick 60b41e40f fix build error
	pick f734508c7 Update tidb-roadmap.md
	pick a3f0f7861 Update tidb-roadmap.md
	pick cda93ef91 Apply suggestions from code review
	edit 2f2396499 remove duplicate
	pick b139606eb Apply suggestions from code review
	pick c3f516dc6 Apply suggestions from code review
	pick 362467e79 Apply suggestions from code review
	```
	
	Save and close the editor.
	
	When the rebase process stops for each commit to be amended, change the commit's author using `git commit --amend --author="Sam Dillard <sam@pingcap.com>"` and continue the rebase process using `git rebase --continue`. The following is an example:
	
	```bash
	➜  docs git:(update-tidb-roadmap.md) git rebase -i 6f9ef141791c8b6becf4ba32d1ffcfd428a744f6
	Stopped at 3a20a0653...  update tidb-roadmap.md -- first draft
	You can amend the commit now, with
	
	  git commit --amend '-S'
	
	Once you are satisfied with your changes, run
	
	  git rebase --continue
	➜  docs git:(3a20a0653) git commit --amend --author="Sam Dillard <sam@pingcap.com>"
	[detached HEAD 6c9f03b4c] update tidb-roadmap.md -- first draft
	 Author: Sam Dillard <sam@pingcap.com>
	 Date: Wed Oct 25 21:06:59 2023 -0700
	 1 file changed, 39 insertions(+), 78 deletions(-)
	➜  docs git:(6c9f03b4c) git rebase --continue
	Stopped at 2f2396499...  remove duplicate
	You can amend the commit now, with
	
	  git commit --amend '-S'
	
	Once you are satisfied with your changes, run
	
	  git rebase --continue
	➜  docs git:(c10d9c25e) git commit --amend --author="Sam Dillard <sam@pingcap.com>"
	[detached HEAD ddbdb71bf] remove duplicate
	 Author: Sam Dillard <sam@pingcap.com>
	 Date: Mon Nov 6 12:02:08 2023 -0800
	 1 file changed, 4 deletions(-)
	➜  docs git:(ddbdb71bf) git rebase --continue
	Successfully rebased and updated refs/heads/update-tidb-roadmap.md.
	➜  docs git:(update-tidb-roadmap.md)
	```
	
	Then, you need to force push the changes:
	
	```bash
	git push --force
	```

**Pull request merged**